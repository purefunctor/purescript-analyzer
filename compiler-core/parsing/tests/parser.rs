use test_each_file::test_each_file;

test_each_file! { in "./compiler-core/parsing/tests/parser" => |content: &str| {
    let lexed = lexing::lex(content);
    let tokens = lexing::layout(&lexed);
    let (parsed, errors) = parsing::parse(&lexed, &tokens);
    let node = parsed.syntax_node();
    insta::assert_debug_snapshot!((node, errors));
}}

test_each_file! { in "./compiler-core/parsing/tests/parser" as lossless => |content: &str| {
    let lexed = lexing::lex(content);
    let tokens = lexing::layout(&lexed);
    let (parsed, _) = parsing::parse(&lexed, &tokens);
    let node = parsed.syntax_node();
    assert_eq!(node.to_string(), content);
}}

test_each_file! { in "./compiler-core/parsing/tests/parser" as stability => |content: &str| {
    let lexed = lexing::lex(content);
    for index in 0..lexed.len() - 1 {
        let partial = lexed.text_in_range(0..index + 1);
        let lexed = lexing::lex(partial);
        let tokens = lexing::layout(&lexed);
        let (parsed, _) = parsing::parse(&lexed, &tokens);
        let node = parsed.syntax_node();
        assert_eq!(node.to_string(), partial);
    }
}}
