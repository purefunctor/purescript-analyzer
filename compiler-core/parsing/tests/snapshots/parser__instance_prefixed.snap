---
source: compiler-core/parsing/tests/parser.rs
expression: "(node, errors)"
---
(
    Module@0..55
      ModuleHeader@0..29
        MODULE@0..6 "module"
        ModuleName@6..23
          Annotation@6..7
            TEXT@6..7 " "
          UPPER@7..23 "InstancePrefixed"
        Annotation@23..24
          TEXT@23..24 " "
        WHERE@24..29 "where"
      LAYOUT_START@29..29 ""
      ModuleStatements@29..54
        InstanceChain@29..54
          InstanceDeclaration@29..54
            Annotation@29..31
              TEXT@29..31 "\n\n"
            INSTANCE@31..39 "instance"
            InstanceHead@39..54
              Annotation@39..40
                TEXT@39..40 " "
              QualifiedName@40..50
                Qualifier@40..48
                  TEXT@40..48 "Data.Eq."
                UPPER@48..50 "Eq"
              TypeConstructor@50..54
                Annotation@50..51
                  TEXT@50..51 " "
                QualifiedName@51..54
                  UPPER@51..54 "Int"
      LAYOUT_END@54..54 ""
      Annotation@54..55
        TEXT@54..55 "\n"
      END_OF_FILE@55..55 ""
    ,
    [],
)
